packages/kbn-zod-helpers/src/is_valid_date_math.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import * as z from 'zod';
+import { parseScheduleDates } from '@kbn/securitysolution-io-ts-utils';
+
+export function isValidDateMath(input: string, ctx: z.RefinementCtx) {
+  const parsedInput = parseScheduleDates(input);
+  if (parsedInput == null) {
+    ctx.addIssue({
+      code: z.ZodIssueCode.custom,
+      message: 'Failed to parse date-math expression',
+    });
```

Comment:
When this happens and Zod generates a validation error message, will it be clear from this message for what field with what value it "Failed to parse date-math expression"?

Reply:
Yes, it will mention a full property path, like `foo.bar: Failed to parse date-math expression`